fastfuels_core/disc_sampling.py

```diff
@@ -44,18 +41,52 @@ def __init__(
         sample_heights=True,
         crown_profile_model="purves",
     ):
+        # Get tree properties as an array for use in Numba
+        trees = trees.dropna()
+        self.trees = TreePopulation(trees)
 
         # What type of crown profile model to use
         if crown_profile_model not in ["beta", "purves"]:
             raise ValueError(
                 "The crown profile model must be one of the following: 'beta' or 'purves'"
             )
         self.crown_profile_model = crown_profile_model
-
-        # Get tree properties as an array for use in Numba
-        trees = trees.dropna()
-        self.trees = TreePopulation(trees)
-        self.tree_props = self.get_tree_props(self.trees)
+        if self.crown_profile_model == "beta":
+            height = trees["HT"].to_numpy()
+            crown_ratio = trees["CR"].to_numpy()
+            crown_length = height * crown_ratio
+            crown_base_height = height - crown_length
+
+            # create BetaCrownProfile
+            self.beta_model = BetaCrownProfile(
+                trees["SPCD"].to_numpy().astype(str), crown_base_height, crown_length
+            )
+            # update tree properties
+            trees["A"] = self.beta_model.a
+            trees["B"] = self.beta_model.b
+            trees["C"] = self.beta_model.c
+            trees["BETA"] = self.beta_model.beta
```

Comment:
can you assign these to variables a, b, c, and beta instead of adding them to the DataFrame?